### PR TITLE
[WIP] Frontend deposit min share implementation

### PR DESCRIPTION
**Pull request by JosephJang7227**

### CODE REVIEW

File: src/calculations/calculation.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import BigNumber from "bignumber.js";
 import { BigNumberWithConfig, validate } from "./utils";
 import { approximateOutAmount } from "./approximation";
+import { getNormalizedReserves } from "./swapOutAmount";
```
Comment (ypeng-deltafi) [inline src/calculations/calculation.ts:4]: it is generated by `copy-ts-client` command?

Comment (JosephJang7227) [reply src/calculations/calculation.ts:4]: Yes, I am using it in the deposit share calculation

File: src/calculations/calculation.ts [hunk 2]
```diff
@@ -270,3 +272,84 @@ export function calculateOutAmountStableSwap(
...
+
+// Calculate expected output from Deposit multiplied by minCoeff
+// Checks if its normalSwap or stableSwap and adjusts initial splitByRatio accordingly
+export function calculateMinOutAmountDeposit(
+  swapInfo: SwapInfo,
+  baseAmount: BigNumber,
+  quoteAmount: BigNumber,
+  marketPrice: BigNumber,
+  minCoeff: BigNumber,
+  normalSwap: boolean,
```
Comment (ypeng-deltafi) [inline src/calculations/calculation.ts:284]: can we change this parameter to swapType? I think we have another type called SerumSwap

File: src/views/Deposit/Deposit.tsx [hunk 3]
```diff
@@ -445,6 +451,18 @@ const Deposit: React.FC<{ poolAddress?: string }> = (props) => {
...
+
+      let minCoeff = new BigNumber(0.99);
+
+      const { minBaseShare, minQuoteShare } = calculateMinOutAmountDeposit(
+        swapInfo,
+        anchorBnToBn(poolConfig.baseTokenInfo, baseAmount),
+        anchorBnToBn(poolConfig.quoteTokenInfo, quoteAmount),
+        new BigNumber(marketPrice),
+        minCoeff,
+        swapInfo.swapType.stableSwap,
```
Comment (ypeng-deltafi) [inline src/views/Deposit/Deposit.tsx:463]: it doesn't make sense to assgin `swapInfo.swapType.stableSwap` to `normalSwap`

File: src/calculations/calculation.ts [hunk 4]
```diff
@@ -270,3 +272,83 @@ export function calculateOutAmountStableSwap(
...
 }
+
+// Calculate expected output from Deposit multiplied by minCoeff
+// Checks if its normalSwap or stableSwap and adjusts initial splitByRatio accordingly
+export function calculateMinOutAmountDeposit(
+  swapInfo: SwapInfo,
+  baseAmount: BigNumber,
+  quoteAmount: BigNumber,
+  marketPrice: BigNumber,
+  minCoeff: BigNumber,
```
Comment (ypeng-deltafi) [inline src/calculations/calculation.ts:283]: please also update it in the dex repo

**Review by ypeng-deltafi — APPROVED** [review]
please test it with mainnet-test or testnet before merge

**Comment by ypeng-deltafi** [comment]
@JosephJang7227 please remove the `[WIP]` label before merge

**Pull request merged**